aea/registries/base.py

```diff
@@ -225,292 +237,165 @@ def _add_protocol(self, protocol_directory: Path):
         )
 
         # instantiate the protocol manager.
-        protocol = Protocol(protocol_name, serializer, protocol_config)
-        self.register((protocol_name, None), protocol)
+        protocol = Protocol(protocol_config.public_id, serializer, protocol_config)
+        protocol_public_id = PublicId(
+            protocol_config.author, protocol_config.name, protocol_config.version
+        )
+        self.register(protocol_public_id, protocol)
 
 
-class HandlerRegistry(Registry):
-    """This class implements the handlers registry."""
+class ComponentRegistry(
```

Comment:
A `ComponentRegistry` is a registry for `SkillComponentType`. The identifier of a skill component, within the set of components of the same type, is the pair `(SkillId, component_name)`.

Reply:
Notice: we have only one class for every type of registry (namely, `HandlerRegistry`, `BehaviourRegistry`, `TaskRegistry`)

Reply:
Cool!